src/cbc_sdk/errors.py

```diff
@@ -138,7 +142,9 @@ def __str__(self):
             msg += " (No further information provided)"
 
         if self.result:
-            msg += ". {}".format(self.result)
+            msg += f". {self.result}"
+        if self.uri:
+            msg += f" <{self.uri}>"
```

Comment:
Would this produce a more legible format?
```suggestion
        msg = "Received error code {0:d} from API".format(self.error_code)
        if self.uri:
            msg += f" ({self.uri})"
        details = ""
        if self.message:
            details += f" ({self.message})"
        if self.result:
            details += f" ({self.result})"
        if not details:
            details += f" (No further information provided)"
        return msg + details
```

Reply:
As an example this:
```
Received error code 500 from API: { "activation_code" : null, "activation_code_expiry_time" : "2023-06-20T22:08:52.307Z", "ad_group_id" : 0, "appliance_name" : null, "appliance_uuid" : null, "auto_scaling_group_name" : null, "av_ave_version" : null, "av_engine" : null, "av_last_scan_time" : null, "av_master" : false, "av_pack_version" : null, "av_product_version" : null, "av_status" : null, "av_update_servers" : null, "av_vdf_version" : null, "base_device" : null, "cloud_provider_account_id" : null, "cloud_provider_resource_id" : null, "cloud_provider_tags" : null, "cluster_name" : null, "compliance_status" : "NOT_ASSESSED", "current_sensor_policy_name" : "Standard", "datacenter_name" : null, "deployment_type" : "WORKLOAD", "deregistered_time" : null, "device_meta_data_item_list" : [ ], "device_owner_id" : 17906278, "email" : "cbc-20230613220541416197-jntial-query-dev01-273", "esx_host_name" : null, "esx_host_uuid" : null, "first_name" : null, "golden_device" : null, "golden_device_id" : null, "id" : 18855350, "infrastructure_provider" : "NONE", "last_contact_time" : "2023-06-13T22:09:59.571Z", "last_device_policy_changed_time" : null, "last_device_policy_requested_time" : "2023-06-13T22:10:03.746Z", "last_external_ip_address" : "66.170.99.2", "last_internal_ip_address" : "192.168.48.106", "last_location" : "UNKNOWN", "last_name" : null, "last_policy_updated_time" : null, "last_reported_time" : "2023-06-13T22:09:59.689Z", "last_reset_time" : null, "last_shutdown_time" : null, "linux_kernel_version" : null, "login_user_name" : null, "mac_address" : "005056a5f270", "middle_name" : null, "name" : "cbc-20230613220541416197-jntial-query-dev01-273", "nsx_distributed_firewall_policy" : null, "nsx_enabled" : null, "organization_id" : 6445430, "organization_name" : "test-e2e-auditremediation.org", "os" : "LINUX", "os_version" : "RHEL 8.4 x64", "passive_mode" : false, "policy_id" : 20389918, "policy_name" : "Standard", "policy_override" : false, "quarantined" : false, "registered_time" : "2023-06-13T22:08:52.347Z", "scan_last_action_time" : null, "scan_last_complete_time" : null, "scan_status" : null, "sensor_gateway_url" : null, "sensor_gateway_uuid" : null, "sensor_kit_type" : "RHEL", "sensor_out_of_date" : true, "sensor_pending_update" : false, "sensor_states" : [ "LIVE_RESPONSE_DISABLED", "LIVE_RESPONSE_NOT_KILLED", "ACTIVE" ], "sensor_version" : "2.14.1.1646819", "status" : "REGISTERED", "target_priority" : "MEDIUM", "uninstall_code" : "I5UYVIPJ", "vcenter_host_url" : null, "vcenter_name" : null, "vcenter_uuid" : null, "vdi_base_device" : null, "vdi_provider" : "NONE", "virtual_machine" : true, "virtual_private_cloud_id" : null, "virtualization_provider" : "VMW_ESX", "vm_ip" : null, "vm_name" : null, "vm_uuid" : null, "vulnerability_score" : 0.0, "vulnerability_severity" : null, "windows_platform" : null } <https://defense-dev01.cbdtest.io/appservices/v6/orgs/3X2TQDYW5V/devices/18855350>
```
could become:
```
Received error code 500 from API (https://defense-dev01.cbdtest.io/appservices/v6/orgs/3X2TQDYW5V/devices/18855350) ({ "activation_code" : null, "activation_code_expiry_time" : "2023-06-20T22:08:52.307Z", "ad_group_id" : 0, "appliance_name" : null, "appliance_uuid" : null, "auto_scaling_group_name" : null, "av_ave_version" : null, "av_engine" : null, "av_last_scan_time" : null, "av_master" : false, "av_pack_version" : null, "av_product_version" : null, "av_status" : null, "av_update_servers" : null, "av_vdf_version" : null, "base_device" : null, "cloud_provider_account_id" : null, "cloud_provider_resource_id" : null, "cloud_provider_tags" : null, "cluster_name" : null, "compliance_status" : "NOT_ASSESSED", "current_sensor_policy_name" : "Standard", "datacenter_name" : null, "deployment_type" : "WORKLOAD", "deregistered_time" : null, "device_meta_data_item_list" : [ ], "device_owner_id" : 17906278, "email" : "cbc-20230613220541416197-jntial-query-dev01-273", "esx_host_name" : null, "esx_host_uuid" : null, "first_name" : null, "golden_device" : null, "golden_device_id" : null, "id" : 18855350, "infrastructure_provider" : "NONE", "last_contact_time" : "2023-06-13T22:09:59.571Z", "last_device_policy_changed_time" : null, "last_device_policy_requested_time" : "2023-06-13T22:10:03.746Z", "last_external_ip_address" : "66.170.99.2", "last_internal_ip_address" : "192.168.48.106", "last_location" : "UNKNOWN", "last_name" : null, "last_policy_updated_time" : null, "last_reported_time" : "2023-06-13T22:09:59.689Z", "last_reset_time" : null, "last_shutdown_time" : null, "linux_kernel_version" : null, "login_user_name" : null, "mac_address" : "005056a5f270", "middle_name" : null, "name" : "cbc-20230613220541416197-jntial-query-dev01-273", "nsx_distributed_firewall_policy" : null, "nsx_enabled" : null, "organization_id" : 6445430, "organization_name" : "test-e2e-auditremediation.org", "os" : "LINUX", "os_version" : "RHEL 8.4 x64", "passive_mode" : false, "policy_id" : 20389918, "policy_name" : "Standard", "policy_override" : false, "quarantined" : false, "registered_time" : "2023-06-13T22:08:52.347Z", "scan_last_action_time" : null, "scan_last_complete_time" : null, "scan_status" : null, "sensor_gateway_url" : null, "sensor_gateway_uuid" : null, "sensor_kit_type" : "RHEL", "sensor_out_of_date" : true, "sensor_pending_update" : false, "sensor_states" : [ "LIVE_RESPONSE_DISABLED", "LIVE_RESPONSE_NOT_KILLED", "ACTIVE" ], "sensor_version" : "2.14.1.1646819", "status" : "REGISTERED", "target_priority" : "MEDIUM", "uninstall_code" : "I5UYVIPJ", "vcenter_host_url" : null, "vcenter_name" : null, "vcenter_uuid" : null, "vdi_base_device" : null, "vdi_provider" : "NONE", "virtual_machine" : true, "virtual_private_cloud_id" : null, "virtualization_provider" : "VMW_ESX", "vm_ip" : null, "vm_name" : null, "vm_uuid" : null, "vulnerability_score" : 0.0, "vulnerability_severity" : null, "windows_platform" : null })
```
or
```
Received error code 500 from API (https://defense-dev01.cbdtest.io/appservices/v6/orgs/3X2TQDYW5V/devices/18855350) (Internal Server Error)
```
or
```
Received error code 500 from API (https://defense-dev01.cbdtest.io/appservices/v6/orgs/3X2TQDYW5V/devices/18855350) (No further information provided)
```

Reply:
@rmcqueen-cb I'm implementing that, with one additional proviso suggested by @avanbrunt-cb: if `result` is a string greater than 100 characters, it will be truncated, with a `[...]` added to the end. That might also help readability in the case of a really big result.